File: Task 2/Java/app/src/main/java/dev/ogabek/java/adapter/NoteAdapter.java
package dev.ogabek.java.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import dev.ogabek.java.R;
import dev.ogabek.java.model.Note;

public class NoteAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    Context context;
    List<Note> notes;

    public NoteAdapter(Context context, List<Note> notes) {
        this.context = context;
        this.notes = notes;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_note, parent, false);
        return new NoteViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        Note note = notes.get(position);

        if (holder instanceof NoteViewHolder) {
            ((NoteViewHolder) holder).text.setText(note.getText());
        }

    }

    @Override
    public int getItemCount() {
        return notes.size();
    }

    private static class NoteViewHolder extends RecyclerView.ViewHolder {

        TextView text;

        public NoteViewHolder(View view) {
            super(view);

            text = view.findViewById(R.id.text);

        }
    }
}

File: Task 2/Kotlin/app/src/main/java/dev/ogabek/kotlin/adapter/NoteAdapter.kt
package dev.ogabek.kotlin.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.model.Note

class NoteAdapter(val context: Context, val notes: ArrayList<Note>): RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_note, parent, false)
        return NoteViewHolder(view)
    }

    class NoteViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val text: TextView = view.findViewById(R.id.text)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val note = notes[position]

        if (holder is NoteViewHolder) {
            holder.apply {
                text.text = note.text
            }
        }

    }

    override fun getItemCount() = notes.size
}